[Feature]: Add LiteLLM Type stubs for mypy

### The Feature

Request from @nsbradford on this

### Motivation, pitch

```shell
-
```


### Twitter / LinkedIn details

_No response_

A quick thing you can do is simply create a file called `litellm/py.typed`.  The contents don't matter, it's a marker file.

I get this error when I run `mypy app.py`:

```
app.py:17: error: Skipping analyzing "litellm.utils": module is installed, but missing library stubs or py.typed marker  [import-untyped]
app.py:17: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

Which I was able to workaround with:

```bash
touch venv/lib/python3.12/site-packages/litellm/py.typed
```

A better method is to generate library stubs,  `*.pyi`.  They speed up mypy analysis.  They can be generated with `stubgen`, but I've not done it before.

--- a/litellm/py.typed
+++ b/litellm/py.typed
@@ -1,0 +1,2 @@
+# Marker file to instruct type checkers to look for inline type annotations in this package.
+# See PEP 561 for more information.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,6 +5,10 @@ description = "Library to easily interface with LLM API providers"
 authors = ["BerriAI"]
 license = "MIT"
 readme = "README.md"
+packages = [
+    { include = "litellm" },
+    { include = "litellm/py.typed"},
+]
 
 [tool.poetry.urls]
 homepage = "https://litellm.ai"

